crates/ark/src/interface.rs

```diff
@@ -453,6 +469,38 @@ impl RMain {
         }
     }
 
+    /// Wait for complete R initialization
+    ///
+    /// Wait for R being ready to take input (i.e. `ReadConsole` was called).
+    /// Resolves as the same time as the `Bus<KernelInfo>` init channel does.
+    ///
+    /// Thread-safe.
+    pub fn wait_r_initialized() {
+        R_INIT.wait();
+    }
+
+    /// Has the `RMain` singleton completed initialization.
+    ///
+    /// This can return true when R might still not have finished starting up.
+    /// See `wait_r_initialized()`.
+    ///
+    /// Thread-safe. But note you can only get access to the singleton on the R
+    /// thread.
+    pub fn is_initialized() -> bool {
+        R_MAIN_INIT.get().is_some()
+    }
+
+    /// Wait for `RMain` singleton initialization
+    ///
+    /// Note that R might still not have finished starting up.
+    /// See `wait_r_initialized()`.
+    ///
+    /// Thread-safe. But note you can only get access to the singleton on the R
+    /// thread.
+    pub fn wait_initialized() {
+        R_MAIN_INIT.wait();
+    }
```

Comment:
I am noting that currently we don't use this

Reply:
I added it for #506, though we eventually determined we could just assume R and RMain are both initialized. I think it makes sense to keep it in the API.

Reply:
After writing this documentation:

```rust
    /// Thread-safe, but note:
    /// - Shouldn't be called from the R thread. If you're on the R thread, it
    ///   means that either R is initialized, or something is very wrong and waiting
    ///   would cause a deadlock anyway.
    /// - You can wait for `RMAIN` to be initialized in another thread, but you
    ///   can only get access to the singleton on the R thread.
```

I decided to remove it from the API 🤣